components/restapi/core/accounts.go

```diff
@@ -164,143 +97,35 @@ func rewardsByOutputID(c echo.Context) (*api.ManaRewardsResponse, error) {
-		claimingEpoch := apiForSlot.TimeProvider().EpochFromSlot(futureBoundedSlotIndex)
-
-		stakingPoolValidatorAccountID = accountOutput.AccountID
-		// check if the account is a validator
-		reward, firstRewardEpoch, lastRewardEpoch, err = deps.Protocol.Engines.Main.Get().SybilProtection.ValidatorReward(
-			stakingPoolValidatorAccountID,
-			stakingFeature,
-			claimingEpoch,
-		)
-
-	case iotago.OutputDelegation:
-		//nolint:forcetypeassert
-		delegationOutput := utxoOutput.Output().(*iotago.DelegationOutput)
-		delegationEnd := delegationOutput.EndEpoch
-		futureBoundedSlotIndex := slotIndex + apiForSlot.ProtocolParameters().MinCommittableAge()
-		claimingEpoch := apiForSlot.TimeProvider().EpochFromSlot(futureBoundedSlotIndex)
-
-		// If Delegation ID is zeroed, the output is in delegating state, which means its End Epoch is not set and we must use the
-		// "last epoch" for the rewards calculation.
-		// In this case the calculation must be consistent with the rewards calculation at execution time, so a client can specify
-		// a slot index explicitly, which should be equal to the slot it uses as the commitment input for the claiming transaction.
-		if delegationOutput.DelegationID.Empty() {
-			delegationEnd = claimingEpoch - iotago.EpochIndex(1)
-		}
-
-		stakingPoolValidatorAccountID = delegationOutput.ValidatorAddress.AccountID()
-
-		reward, firstRewardEpoch, lastRewardEpoch, err = deps.Protocol.Engines.Main.Get().SybilProtection.DelegatorReward(
-			stakingPoolValidatorAccountID,
-			delegationOutput.DelegatedAmount,
-			delegationOutput.StartEpoch,
-			delegationEnd,
-			claimingEpoch,
-		)
-	}
-	if err != nil {
-		return nil, ierrors.Wrapf(echo.ErrInternalServerError, "failed to calculate reward for output %s: %s", outputID.ToHex(), err)
-	}
-
-	latestCommittedEpochPoolRewards, poolRewardExists, err := deps.Protocol.Engines.Main.Get().SybilProtection.PoolRewardsForAccount(stakingPoolValidatorAccountID)
-
-	if err != nil {
-		return nil, ierrors.Wrapf(echo.ErrInternalServerError, "failed to retrieve pool rewards for account %s: %s", stakingPoolValidatorAccountID.ToHex(), err)
-	}
-	if !poolRewardExists {
-		latestCommittedEpochPoolRewards = 0
-	}
-
-	return &api.ManaRewardsResponse{
-		StartEpoch:                      firstRewardEpoch,
-		EndEpoch:                        lastRewardEpoch,
-		Rewards:                         reward,
-		LatestCommittedEpochPoolRewards: latestCommittedEpochPoolRewards,
-	}, nil
+	return deps.RequestHandler.RewardsByOutputID(outputID, slot)
 }
 
 func selectedCommittee(c echo.Context) (*api.CommitteeResponse, error) {
-	timeProvider := deps.Protocol.CommittedAPI().TimeProvider()
-
-	var slot iotago.SlotIndex
+	timeProvider := deps.RequestHandler.CommittedAPI().TimeProvider()
 
 	epoch, err := httpserver.ParseEpochQueryParam(c, api.ParameterEpoch)
 	if err != nil {
 		// by default we return current epoch
-		slot = timeProvider.SlotFromTime(time.Now())
+		slot := timeProvider.SlotFromTime(time.Now())
```

Comment:
Maybe we should add `CurrentSlot` and `CurrentEpoch` to the timeProvider?

Reply:
That would seem like a nice addition, but I wouldn't add it in this PR.